app/Components/Sudoku Board/SudokuBoard.tsx

```diff
@@ -102,10 +113,7 @@ const SudokuBoard = (props: SudokuBoardProps) => {
     if (sudokuBoard.selectedCell == null) {
       return;
     }
-    const currentSelectedCell: CellProps | null = getCurrentSelectedCell();
-    if (currentSelectedCell == null) {
-      return;
-    }
+    const currentSelectedCell = getCurrentSelectedCell() as CellProps;
```

Comment:
While this is definitely an improvement I think it would be easier to read if you did this before the null check then return if currentSelectedCell is equal to null instead of sudokuBoard.selectedCell.